editor/src/clj/editor/gui.clj

```diff
@@ -1480,24 +1496,25 @@
                                                                     layer-index (assoc :layer-index layer-index)))))))))
   (output gpu-texture TextureLifecycle (g/constantly nil))
-  (output aabb g/Any :cached (g/fnk [source-scene]
-                               (if (some? source-scene)
-                                 (:aabb source-scene)
-                                 geom/empty-bounding-box)))
-  (output scene g/Any :cached (g/fnk [_node-id id aabb transform source-scene scene-children color+alpha inherit-alpha]
-                                     (let [scene (if source-scene
-                                                   (let [updatable (assoc (:updatable source-scene) :node-id _node-id)]
-                                                     (cond-> (scene/claim-scene source-scene _node-id id)
-                                                       updatable
-                                                       ((partial scene/map-scene #(assoc % :updatable updatable)))))
-                                                   {:renderable {:passes [pass/selection]
-                                                                 :user-data {:color color+alpha :inherit-alpha inherit-alpha}}})]
-                                       (-> scene
-                                           (assoc
-                                             :node-id _node-id
-                                             :aabb aabb
-                                             :transform transform)
-                                           (update :children (fnil into []) scene-children)))))
+  (output aabb g/Any (g/fnk [source-scene]
+                       (if (some? source-scene)
+                         (:aabb source-scene)
+                         geom/empty-bounding-box)))
+  (output scene g/Any :cached (g/fnk [_node-id aabb transform source-scene scene-children color+alpha inherit-alpha visible enabled]
+                                (let [scene (if source-scene
+                                              (if-some [updatable (some-> source-scene :updatable (assoc :node-id _node-id))]
+                                                (scene/map-scene #(assoc % :updatable updatable) source-scene)
+                                                source-scene)
+                                              {:renderable {:passes [pass/selection]
+                                                            :user-data {:color color+alpha :inherit-alpha inherit-alpha}}})]
+                                  (-> scene
+                                      (assoc
+                                        :node-id _node-id
+                                        :aabb aabb
+                                        :transform transform
+                                        :visible-self? (and visible enabled)
+                                        :visible-children? enabled)
```

Comment:
I would suggest renaming `:visible-self?` to `:visible-self` and `:visible-children?` to `:visible-children`. I explained my reasoning here: https://vlaaad.github.io/2019-03-30/question-marks-in-clojure. 
In short: question mark suffixes are for predicates only, but even though keywords are functions, they are first and foremost data identifiers. They are frequently used in destructuring, which will result in local bindings like `visible-self?` that look like predicates but are in fact booleans.

Reply:
I agree with the sentiment, but the map already used question marks for other keys like `:topmost?` and I didn't want to pollute the PR by renaming the keys, so I followed suite here for consistency.

Reply:
OK, I'm on board. Let's take that approach from now on. But for this one, I'll do a separate PR later to fix all of them once this reaches `editor-dev`. I don't want to trigger a new CI run at this point. 😜

Reply:
Sounds reasonable 👍